### ISSUE TITLE
Regarding the model computational cost issue

### ISSUE THREAD

**Issue by yunmuz** [zizheng-guo/RhythmMamba#14]
I'm trying to use the following code to obtain specific computational cost metrics, such as parameter count, FLOPs, or FPS, but the results are not as expected. I would appreciate your help in analyzing the issue and suggesting optimizations or alternative approaches. Here is the code:
import torch
import time
import numpy as np
from fvcore.nn import FlopCountAnalysis
from neural_methods.model.RhythmMamba import RhythmMamba

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
model = RhythmMamba().to(device)
model.eval()

input_tensor = torch.randn(1, 160, 3, 128, 128).to(device)
num_frames = 30

# ---------------------------

total_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
print(f"Total Parameters: {total_params / 1e6:.3f} M")

# ---------------------------

with torch.no_grad():
    flops = FlopCountAnalysis(model, input_tensor)
    total_macs = flops.total()
    macs_per_frame = total_macs / num_frames
    print(f"MACs per frame: {macs_per_frame / 1e9:.3f} GMACs")

# ---------------------------
with torch.no_grad():
    for _ in range(10):  # warm-up
        _ = model(input_tensor)

    start = time.time()
    with torch.no_grad():
        _ = model(input_tensor)
    end = time.time()

    inference_time = end - start
    fps = num_frames / inference_time
    print(f" Throughput: {fps:.2f} frames/sec")

# ---------------------------

torch.cuda.reset_peak_memory_stats()
with torch.no_grad():
    _ = model(input_tensor)
peak_memory = torch.cuda.max_memory_allocated(device=device) / (1024 ** 2)
memory_per_frame = peak_memory / num_frames
print(f" Peak GPU Memory per frame: {memory_per_frame:.2f} MB")

![Image](https://github.com/user-attachments/assets/8ba7f057-6dc9-405e-b7f6-b8d99e01c6cd)

**Comment by yunmuz** [comment]
@zizheng-guo

**Comment by yunmuz** [comment]
@zizheng-guo

**Comment by zizheng-guo** [comment]
I have open-sourced the code related to computational cost. You can refer to it here: [evaluation/cost_cal.md](https://github.com/zizheng-guo/RhythmMamba/blob/main/evaluation/cost_cal.md)
